crates/schema-impl/src/schemas_impl.rs

```diff
@@ -197,12 +197,13 @@ impl ServiceSchemas {
     ) -> HashMap<String, MethodSchemas> {
         svc_desc
             .methods()
+            .filter(|descriptor| method_meta.contains_key(&descriptor.name().to_string()))
```

Comment:
I think the posted snippet does what it should do (computing the right `Vec<SchemasUpdateCommand>`). The problem is rather the `SchemasInner::apply_updates` which calls the `compute_service_methods`. I believe you are right that the method needs to check whether `method_meta` actually contains the `descriptor.name()`. Otherwise we would add all methods of the provided `ServiceDescriptor`.

One way to do it could be:

```
svc_desc
    .methods()
    .flat_map(|descriptor| {
        let method_name = descriptor.name().to_string();
        method_meta.get(&method_name).map(|metadata| {
            (
                method_name,
                MethodSchemas::new(descriptor, metadata.input_fields_annotations.clone()),
            )
        })
    })
    .collect()
```

Reply:
If you do this change, please also update the test `register_issue682` to register the `Issue682` method instead of the `Greet` method.

Reply:
I won't apply this change now. This bad mocking thing will go away once I get #981 done.